akvo-top-asn

```diff
@@ -35,22 +34,39 @@ parser.add_argument('-c', '--config', type=check_conf_filename,
                     default='akvo-top-asn.conf', help='configuration filename')
 parser.add_argument(
     '--starttime',
-    type=check_time_stamp, default=datetime.datetime.utcnow()-datetime.timedelta(days=7)
+    type=check_time_stamp 
 )
 parser.add_argument(
     '--endtime',
-    type=check_time_stamp, default=datetime.datetime.utcnow()
+    type=check_time_stamp 
 )
 
-# parse config file
+parser.add_argument("--days", type=int, default=0)
+parser.add_argument("--weeks", type=int, default=0)
+parser.add_argument("--months", type=int, default=0)
+
 args = parser.parse_args()
 config = configparser.ConfigParser()
 config.read(args.config)
 
-time_range = [args.starttime.strftime(
-    '%Y-%m-%d %H:%M:%S'), args.endtime.strftime('%Y-%m-%d %H:%M:%S')]
-directions = ['Out', 'In']
+starttime = args.starttime
+endtime = args.endtime
+
+# if no start time is defined we will take the current time
+if endtime is None:
+    endtime = datetime.datetime.utcnow().isoformat()
+
+if starttime is None:
+    delta = datetime.timedelta(
+        days=args.days + args.weeks * 7 + args.months * 30,
+    )
+
+    starttime = (datetime.datatime.utcnow() - delta).isoformat()
+
+
+time_range = [args.starttime, args.endtime]
```

Comment:
This breaks clickhouse if subseconds are specified - the implicit stringification relies on isoformat which may contains subseconds, so `strftime` seems to be required.